### PR TITLE
Add Introduction to Ruby Programming Section and Lesson

### PR DESCRIPTION
**Pull request by KevinMulhern**
This adds a new "Introduction" section and lesson that will be taken at the start of the Ruby course.
We will eventually move the installation instructions for Ruby out of Web Development 101 and into this
section.

### CODE REVIEW

File: ruby_programming/introduction/lesson_introduction_to_ruby_programming.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+### Learning Ruby before Ruby on Rails
+Some people believe you can just dive right into Rails and start firing out websites.  Rails is a framework built using Ruby and every piece of code in it is Ruby.  When (not *if*) something in your project breaks, you'd better be able to debug it. And what happens when you want to stretch your wings and do something just a bit beyond what the vanilla tutorials show you how to do? The amount of time you'd spend googling your error messages and staring blankly at help docs was better spent learning Ruby.
+
+### Break Down What You'll Learn
+Ruby's a big language so it's been broken up into smaller chunks to make it more digestible. Some of the early material will be fairly straightforward and will rely on simple exercises to help reinforce understanding. But the goal is for you to get a much deeper and more practical understanding of the material than that. As we get further along and into some of the more advanced topics, we'll be learning less and building more... just the way it should be.
```
Comment (leosoaivan) [inline ruby_programming/introduction/lesson_introduction_to_ruby_programming.md]: more...just

File: ruby_programming/introduction/lesson_introduction_to_ruby_programming.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+* In the first section you'll learn all about the basics of Ruby and programming itself.
+* In the second section section you will learn about Object Oriented Programming, which will show you how to organize your code into reusable objects.
+* In the third section you will dive into a bit of basic computer science, where you'll learn about recursion and common data structures.
+* In the fourth section, you will learn the basics of test driven development.
```
Comment (leosoaivan) [inline ruby_programming/introduction/lesson_introduction_to_ruby_programming.md]: test-driven

File: ruby_programming/introduction/lesson_introduction_to_ruby_programming.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+### Break Down What You'll Learn
+Ruby's a big language so it's been broken up into smaller chunks to make it more digestible. Some of the early material will be fairly straightforward and will rely on simple exercises to help reinforce understanding. But the goal is for you to get a much deeper and more practical understanding of the material than that. As we get further along and into some of the more advanced topics, we'll be learning less and building more... just the way it should be.
+
+* In the first section you'll learn all about the basics of Ruby and programming itself.
+* In the second section section you will learn about Object Oriented Programming, which will show you how to organize your code into reusable objects.
```
Comment (leosoaivan) [inline ruby_programming/introduction/lesson_introduction_to_ruby_programming.md]: Object-Oriented

**Comment by leosoaivan** [comment]
@leila-alderman would you like to give this a once over as well?

**Comment by leila-alderman** [comment]
I'm just going to fully admit my Git limitations here and say that I'll edit these lessons after this PR gets merged. I don't think that anyone wants to address all of my suggested edits through individual comments. :)

**Comment by leosoaivan** [comment]
Works for me. Merged for you to edit @leila-alderman.

**Pull request merged**